client/service/src/client/wasm_overwrite.rs

```diff
@@ -0,0 +1,140 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+use std::{ 
+	fs, collections::{HashMap, hash_map::DefaultHasher}, path::Path,
+	hash::Hasher as _, 
+};
+use codec::{Encode, Decode};
+use sp_core::{
+	traits::FetchRuntimeCode,
+};
+use sp_state_machine::{BasicExternalities, backend};
+use sp_blockchain::Result;
+use sc_executor::RuntimeInfo;
+use sp_version::RuntimeVersion;
+use sp_core::traits::RuntimeCode;
+use hash_db::Hasher;
+
+#[derive(Clone, Debug)]
+struct WasmBlob {
+	code: Vec<u8>,
+}
+
+impl WasmBlob {
+	fn new(code: Vec<u8>) -> Self {
+		Self { code }
+	}
+
+	fn runtime_code(&self, heap_pages: Option<u64>) -> RuntimeCode {
+		RuntimeCode {
+			code_fetcher: self,
+			hash: make_hash(self.code.as_slice()).encode(),
+			heap_pages,
+		}
+	}
+}
+
+/// Make a hash out of a byte string using the default rust hasher
+pub fn make_hash<K: std::hash::Hash + ?Sized>(val: &K) -> Vec<u8> {
+	let mut state = DefaultHasher::new();
+	val.hash(&mut state);
+	state.finish().to_le_bytes().to_vec()
+}
+
+impl FetchRuntimeCode for WasmBlob {
+	fn fetch_runtime_code<'a>(&'a self) -> Option<std::borrow::Cow<'a, [u8]>> {
+		Some(self.code.as_slice().into())
+	}
+}
+
+#[derive(Clone, Debug)]
+pub struct WasmOverwrite<E> {
+	// Map of runtime spec version -> Wasm Blob
+	overwrites: HashMap<u32, WasmBlob>,
+	executor: E,
+	enabled: bool,
+}
+
+impl<E> WasmOverwrite<E> 
+where 
+	E: RuntimeInfo + Clone + 'static 
+{
+	pub fn new<P>(path: P, enabled: bool, executor: E) -> Result<Self> 
+	where
+		P: AsRef<Path>,
+	{
+		let overwrites = Self::scrape_overwrites(path.as_ref(), &executor)?;
```

Comment:
If we go with `enabled` flag, then this line should probably be guaraded on that flag.

Reply:
changed to one cli arg